e2e/machine_migration_test.go

```diff
@@ -0,0 +1,370 @@
+}
+
+func verifyMachineAuthoritative(ctx context.Context, cl client.Client, machineName string, auth machinev1beta1.MachineAuthority) {
+	By(fmt.Sprintf("Verify the Machine authoritative is %s", auth))
+	Eventually(func() machinev1beta1.MachineAuthority {
+		mapiMachine, err := mapiframework.GetMachine(cl, machineName)
+		if err != nil {
+			return ""
+		}
+		return mapiMachine.Status.AuthoritativeAPI
+	}, framework.WaitMedium, framework.RetryMedium).Should(
+		Equal(auth),
+		"MAPI Machine status.AuthoritativeAPI should be %s", auth)
+}
+
+func verifyMachineSynchronizedCondition(ctx context.Context, cl client.Client, machineName string, auth machinev1beta1.MachineAuthority) {
+	By("Verify the MAPI Machine synchronized condition is True")
+	var expectedMessage string
+	switch auth {
+	case machinev1beta1.MachineAuthorityMachineAPI:
+		expectedMessage = "Successfully synchronized MAPI Machine to CAPI"
+	case machinev1beta1.MachineAuthorityClusterAPI:
+		expectedMessage = "Successfully synchronized CAPI Machine to MAPI"
+	default:
+		Fail(fmt.Sprintf("unknown authoritative API type: %v", auth))
+	}
+
+	Eventually(func() []machinev1beta1.Condition {
+		mapiMachine, err := mapiframework.GetMachine(cl, machineName)
+		if err != nil {
+			return nil
+		}
+		return mapiMachine.Status.Conditions
+	}, framework.WaitMedium, framework.RetryMedium).Should(
+		ContainElement(
+			SatisfyAll(
+				HaveField("Type", Equal(SynchronizedCondition)),
+				HaveField("Status", Equal(corev1.ConditionTrue)),
+				HaveField("Reason", Equal("ResourceSynchronized")),
+				HaveField("Message", Equal(expectedMessage)),
+			),
+		),
+		fmt.Sprintf("Expected Synchronized condition for %s not found or incorrect", auth))
+}
+
+func verifyMachineSynchronizedGeneration(ctx context.Context, cl client.Client, machineName string, auth machinev1beta1.MachineAuthority) {
+	synchronizedGeneration := -1
+	Eventually(func() int {
+		By("Get the MAPI Machine synchronizedGeneration")
+		mapiMachine, err := mapiframework.GetMachine(cl, machineName)
+		if err != nil {
+			return 0
+		}
+		synchronizedGeneration = int(mapiMachine.Status.SynchronizedGeneration)
+		mapiGeneration := int(mapiMachine.Generation)
+		switch auth {
+		case machinev1beta1.MachineAuthorityClusterAPI:
+			By("Return the CAPI Machine generation")
+			capiMachine, err := framework.GetMachine(cl, machineName, framework.CAPINamespace)
+			if err != nil {
+				return 0
+			}
+			return int(capiMachine.Generation)
+		case machinev1beta1.MachineAuthorityMachineAPI:
+			By("Return the MAPI Machine generation")
+			return mapiGeneration
+		default:
+			Fail(fmt.Sprintf("unknown authoritative API type: %v", auth))
+			return 0
+		}
+
+	}, framework.WaitShort, framework.RetryShort).Should(Equal(synchronizedGeneration), "synchronizedGeneration is not %s  machine's generation", auth)
+}
+
+func verifyMAPIMachinePausedCondition(ctx context.Context, cl client.Client, machineName string, auth machinev1beta1.MachineAuthority) {
```

Comment:
Dam suggested change auth to authority, see [this](https://github.com/openshift/cluster-capi-operator/pull/330#discussion_r2215693835)